src/frontend/src/utils/iiConnection.ts

```diff
@@ -393,10 +396,16 @@ export class Connection {
     });
   };
 
-  lookupRecovery = async (userNumber: UserNumber): Promise<DeviceData[]> => {
+  lookupRecovery = async (
+    userNumber: UserNumber
+  ): Promise<RecoveryDevice[]> => {
     const actor = await this.createActor();
-    const allDevices = await actor.lookup(userNumber);
-    return allDevices.filter((device) => "recovery" in device.purpose);
+    const allDevices: Omit<DeviceData, "alias">[] = await actor.lookup(
+      userNumber
+    );
+    return allDevices.filter((device): device is RecoveryDevice =>
```

Comment:
doesn't `isRecoveryDevice` already have `is RecoveryDevice`?

Reply:
Yes, but TypeScript is unfortunately not able to use that for type narrowing: https://github.com/microsoft/TypeScript/issues/16069

Reply:
```suggestion
    return allDevices.filter(isRecoveryDevice);
```

wouldn't this work though?

Reply:
Ah right. I thought I tried this, but apparently I had another typing thing mixed in... 🤔 
Thanks!